File: solutions/two_sum.py
class Solution:
    def twoSum(nums: list[int], target: int) -> list[int]:
        numMap = {} # val -> index

        for i, n in enumerate(nums):
            difference = target - n
            

            if difference in numMap:
                return [numMap[difference], i]

            numMap[n] = i

        #for i in range(len(nums)):
            #for j in range(len(nums)):
                #if j == i:
                    #break
                #if nums[i] + nums[j] == target:
                    #return [i,j]

File: solutions/java/twoSum.java
import java.util.HashMap;

class Solution {
	public int[] twoSum(int[] nums, int target) {
		HashMap<Integer, Integer> numMap = new HashMap<>();
		int n = nums.length;

		for (int i = 0; i < n; i++) {
			int diff = target - nums[i];

			if (numMap.containsKey(diff)) {
				return new int[] { numMap.get(diff), i };
			}

			numMap.put(nums[i], i);
		}
		return new int[] {};
	}
}
